scenarios/workflow_on_many_task_queues.go

```diff
@@ -0,0 +1,33 @@
+package scenarios
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/temporalio/omes/loadgen"
+	"github.com/temporalio/omes/loadgen/kitchensink"
+)
+
+func init() {
+	loadgen.MustRegisterScenario(loadgen.Scenario{
+		Description: "Each iteration executes a single workflow on one of the task queues. " +
+			"Workers must be started with --task-queue-suffix-index-end as one less than task queue count here. " +
+			"Additional options: task-queue-count (required).",
+		Executor: loadgen.KitchenSinkExecutor{
+			WorkflowParams: kitchensink.NewWorkflowParams(kitchensink.NopActionExecuteActivity),
+			PrepareWorkflowParams: func(ctx context.Context, opts loadgen.RunOptions, params *kitchensink.WorkflowParams) error {
+				// Require task queue count
+				if opts.ScenarioOptionInt("task-queue-count", 0) == 0 {
+					return fmt.Errorf("task-queue-count option required")
+				}
+				return nil
+			},
+			UpdateWorkflowOptions: func(ctx context.Context, run *loadgen.Run, options *loadgen.KitchenSinkWorkflowOptions) error {
```

Comment:
This is roundabout way to do this, I'd use the generic executor and have all of the logic inline instead of mutating the params every iteration.

Reply:
But this comes with all the defaults set if you want to just alter them. But the good part about this approach and similar is you can do either (and we can add other callbacks in the future if we want them).